### PR TITLE
Make distributor log ingester details when a push request fails

### PR DESCRIPTION
**Pull request by duricanikolic**
#### What this PR does
Currently, if an ingester fails to push a request, the distributor that triggered the push loggs an error message like this:
```
msg="push error" err="failed pushing to ingester: <original error>"
```
This message doesn't contain any detail about which ingester was involved. This PR enriches the logged error with ingester's ID, so the improved log message will have the following format:
```
msg="push error" err="failed pushing to ingester <ingester ID>: <original error>"
```

#### Checklist

- [x] Tests updated.
- [na] Documentation added.
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [na] [`about-versioning.md`](/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/distributor/distributor_test.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ import (
 )
 
 var (
-	errFail                    = httpgrpc.Errorf(http.StatusInternalServerError, "Fail")
+	errFail                    = status.Error(codes.Internal, "Fail")
```
Comment (pracucci) [inline pkg/distributor/distributor_test.go:68]: Why this change?

File: pkg/distributor/distributor_test.go [hunk 2]
```diff
@@ -1139,8 +1139,8 @@ func TestDistributor_PushQuery(t *testing.T) {
 				assert.EqualError(t, err, tc.expectedError.Error())
 
 				// Assert that downstream gRPC statuses are passed back upstream
-				_, ok := httpgrpc.HTTPResponseFromError(err)
```
Comment (pracucci) [inline pkg/distributor/distributor_test.go:1142]: Why this change?

File: pkg/distributor/distributor_test.go (hunk 1 above)
Comment (duricanikolic) [reply pkg/distributor/distributor_test.go:68]: `errFail` is used in `mockIngester` struct as an example of error returned by that ingester client. After the recent error handling refactoring, it is expected that ingesters will not return errors with HTTP status codes anymore. This is still the default behaviour, but will soon be deprecated.
As a consequence, I replaced the current `errFail` value with an error with a pure gRPC code. 
It is worth noting that this change could be done in a separate PR.

File: pkg/distributor/distributor_test.go (hunk 2 above)
Comment (duricanikolic) [reply pkg/distributor/distributor_test.go:1142]: This change is related to the change discussed above. 
The previous version of this test was ensuring that the error returned by `distributor.QueryStream()` was produced by using the `httpgrpc` package. This is, anyway, not true. Before error handling improvement, it was the case sometimes, but after error handling improvement it is not the case anymore. The reason why the test was not failing was because we were using `errFail` defined as `httpgrpc.Errorf(http.StatusInternalServerError, "Fail")` as the only error returned by `mockIngester`, and that made the test successful. 
Again, this could be done in a separate PR.

Comment (pracucci) [reply pkg/distributor/distributor_test.go:1142]: I'm fine with the change in this PR, but as a general good practice I recommend to keep each PR focused on a single scope (ideally what the PR title says). As a retrospective, this test fix may have been a dedicated PR, which would have been easier to understand and in the future easier to track when we look back at changes.

_Just a feedback for the next time._
